Helper/ApiData.php

```diff
@@ -267,6 +277,9 @@ public function processApiRequest(
                     'Successful response for payment providers.'
                 );
             } elseif ($requestType === 'add_card') {
+                if ($this->preventAdminActions->isAdminAsCustomer()) {
+                    throw new ValidationException(__('Admin user is not authorized for this operation'));
+                }
```

Comment:
Please execute the validation inside controllers.
Having the ApiData executing a validation for user account unexpected and too deep into the module's callstack. 
This validation should run when a controller is called. not inside the deeper logic in the module.

Reply:
Done